app/validators/user_role_name_validator.rb

```diff
@@ -0,0 +1,9 @@
+class UserRoleNameValidator < ActiveModel::Validator
```

Comment:
Would using a proc work?

`validates_inclusion_of :role, in: ->(roles) { valid_roles }, message: "is not a valid value"`